contracts/aggregators/light-client/src/contract.rs

```diff
@@ -0,0 +1,133 @@
+use cosmwasm_std::{
+    entry_point, to_json_binary, Binary, Deps, DepsMut, Env, MessageInfo, Response, StdResult,
+};
+use cw2::set_contract_version;
+
+use crate::error::{ContractError, ContractResult};
+use crate::msg::{InstantiateMsg, QueryMsg, SudoMsg};
+use crate::state::MERKLE_ROOTS;
+
+// version info for migration info
+const CONTRACT_NAME: &str = "crates.io:slinky-avs-contracts";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+const CACHE_SIZE: usize = 6;
+
+/// sudo is the main entrypoint for the contract. It can only be called by modules.
+/// This function handles:
+///  * the deserialization of the input msg
+///  * aggregation over the VE light client inputs
+///  * updating contract state to store agreed upon state updates
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn sudo(_: DepsMut, _: Env, _: SudoMsg) -> ContractResult<Response> {
+    ContractResult::Ok(Response::new())
+}
+
+/// instantiate is used to construct the contract
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    _: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+    Ok(Response::new()
+        .add_attribute("method", "instantiate")
+        .add_attribute("owner", info.sender))
+}
+
+pub mod execute {
+    use super::*;
+    use crate::state::ChainHashes;
+
+    /// write_merkle_roots implements the state update method of the contract.
+    /// Merkle roots are input using a map of chain_id to hash value.
+    /// If a chain has reached the maximum cache size, it evicts the oldest entry and
+    /// inserts a new one.
+    /// Otherwise, it writes a new vector to state for the chain.
+    pub fn write_merkle_roots(
+        deps: DepsMut,
+        merkle_roots: Vec<(String, Binary)>,
+    ) -> Result<Response, ContractError> {
+        for (chain_id, merkle_hash) in merkle_roots.iter() {
+            // Get the existing vector of merkle roots for the chain_id
+            // let mut root_set: Vec<Binary>;
+            let mut root_set: ChainHashes;
+            if MERKLE_ROOTS.has(deps.storage, chain_id.to_string()) {
+                root_set = MERKLE_ROOTS.load(deps.storage, chain_id.clone()).unwrap();
```

Comment:
Feels like having a map so that u can look up by given merkle root hash in the queries would be nice, but may lose a nice way to handle the pruning so prob not something to do for hackathon but would be something I'd like if this was launched for real